### PR TITLE
2dm enhancements

### PR DESCRIPTION
**Pull request by PeterPetrik**

### CODE REVIEW

**Comment by PeterPetrik** [comment]
fix #325 

fix #218

File: mdal/frmts/mdal_2dm.cpp [hunk 1]
```diff
@@ -50,6 +50,21 @@ bool _parse_vertex_id_gaps( std::map<size_t, size_t> &vertexIDtoIndex, size_t ve
   return false;
 }
 
+void _persist_native_index( std::vector<double> &arr, size_t nativeID, size_t ourId, size_t maxOurId )
```
Comment (vcloarec) [inline mdal/frmts/mdal_2dm.cpp:53]: static ?

**Review by vcloarec — COMMENTED** [review]
looks good.
If there is no gap between native and mdal indexes, the indexes are not exposed as dataset group? In QGIS, how the user can see the indexes?

**Comment by PeterPetrik** [comment]
> If there is no gap between native and mdal indexes, the indexes are not exposed as dataset group? In QGIS, how the user can see the indexes?

i think if there are no gaps, it is simple (feature ID - 1)

**Pull request merged**